source/wifi/wifi_hal.c

```diff
@@ -9504,6 +9504,19 @@ INT wifi_getApAssociatedDevice(INT ap_index, mac_address_t *output_deviceMacAddr
 #else
 INT wifi_getApAssociatedDevice(INT ap_index, CHAR *output_buf, INT output_buf_size)
 {
-     return RETURN_OK;
+    char cmd[128];
+    BOOL status = false;
+
+    if(ap_index > MAX_APS)
+        return RETURN_ERR;
+
+    wifi_getApEnable(ap_index,&status);
+    if (!status)
+        return RETURN_OK;
+
+    sprintf(cmd, "hostapd_cli -i %s%d list_sta | tr -d \"\\n\"", AP_PREFIX, ap_index);
```

Comment:
will there be space separated list of client MAC address in the buffer?

Reply:
No, there is nothing between each MAC address. 
and I find that WiFiAgent use "," to separate MAC address. Should I change my implementation in this way?

Reply:
Yes @BroLeaf . please have comma separated list of MAC addresses in the buffer.

Reply:
Hi @manigandanj ,
Your method would remain the last newline as output, but WiFiAgent seems did not consider the last newline character in buffer.
I think this may cause error when WiFiAgent call this function.
So, I remove the all newline and the last comma as my implementation.
`hostapd_cli -i %s%d list_sta | tr '\n' ',' | sed 's/.$//'`